### ISSUE TITLE
Is it possible to use react-virtualized's WindowScroller (for searching and scrolling to an item)?

### ISSUE THREAD

**Issue by Avivhdr** [diogofcunha/react-virtualized-tree#46]
Is there a way to access _react-virtualized_'s _WindowScroller_ functionality or something like that?
In case i have an _id_ of one of the nodes that were sent to the _Tree_ component, is there a way to use a method that will scroll that node in to view?

Like in this example:
https://bvaughn.github.io/react-virtualized/#/components/WindowScroller

Thanks!

**Comment by diogofcunha** [comment]
Hi @Avivhdr at the moment it isn't possible to do that but it seems a reasonable thing to support, I will tackle that over the weekend.

I think spreading the scroll prop down should be enough

**Comment by Avivhdr** [comment]
Hi @diogofcunha,
With a bit of instructions i would be happy to try and tackle this issue my self.

**Comment by mishaberman** [comment]
I would love this feature as well :) Let me know if you need me to test :)

**Comment by diogofcunha** [comment]
I think it would be a good idea to create a slack or gitter chat and discuss how to do this, we can start just by passing the prop down.

Docs for the List can be found in https://github.com/bvaughn/react-virtualized/blob/master/docs/List.md, I think allowing scrollToIndex as a prop for the tree should be enough. If the logic to get the id becomes too complicated we can just  have a prop named scroolToId and get the index internally based on the flattened nodes.

**Comment by Avivhdr** [comment]
@diogofcunha 
I think _scrollToId_ would be a good idea but maybe it would need to get an array of ids, just in case there would be a duplicate id for a group and a child. the first item would be the group's id, the second item would be the child's id.
And for the gitter/slack idea: _avivhdr@gmail.com_

**Comment by diogofcunha** [comment]
Released as part of version `2.2.0`

Thanks for your contribution @Avivhdr

**Comment by FilipeCosta06** [comment]
Hi @diogofcunha 

Is there any documentation about this feature available somewhere here (that it a nifty feature that should be showcased imo) ?

However, my IntelliSense isn't listing any function function called `scrollToIndex` from the Tree object (which is actually Tree Container). Only `scrollToId` is exposed (and `scrollToAlignment`) but this (potentially) loops over all the data since it's using `_.findIndex()` and that might pose a problem as I will be dealing with big amounts of data.

The nodes have their flattened `index` with them (they receive it from the rowRenderer) and if that is the value returned by `scrollToId`, then all I need is a direct way to `scrollToIndex`. Do you think this could be achieved or should I fork the project for just this (I am already replacing the left margin by a left padding, for UI reasons) or was `scrollToIndex` not exposed by mistake in a recent release ?

Lastly, we get a warning when supplying a string to `scrollToId`, saying it is expecting a number
![image](https://user-images.githubusercontent.com/25252573/84018612-7f9b1780-a980-11ea-8312-2d5ee0d429c0.png)

tl;dr:
- Can you expose `scrollToIndex` ?
- I think there is a PropTypes error on `scrollToId` (expects `number` instead of `string`)
